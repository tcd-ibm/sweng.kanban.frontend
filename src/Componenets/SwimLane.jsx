import React from 'react';
import { Box, Typography } from '@material-ui/core';
import KBCard from './Card';

const SwimLane = ({swimLaneData, swimLaneTitle}) => {
  return (
    <Box sx={{
        display: 'flex',
        alignItems: 'flex-start',
        flexDirection: 'column',
        bgcolor: '#9bbedb',
        p: 1,
        m: 2,
        borderRadius: 1
      }}
    >
      <Typography>{swimLaneTitle}</Typography>
      <Box sx={{
          display: 'flex',
          flexDirection: 'column',
          bgcolor: '#f5a7bb',
          p: 1,
          m: 1,
          borderRadius: 3
        }}
      >
        {swimLaneData?.map(card =>
          <KBCard title={card.taskTitle} desc={card.taskDescription} key={card._id}/>
          )
        }
      </Box>
    </Box>
  )
}
export default SwimLane;