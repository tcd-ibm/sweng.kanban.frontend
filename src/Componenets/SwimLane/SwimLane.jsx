import React from 'react';
import { Box, Typography } from '@material-ui/core';
import KBCard from '../Card/Card';

const SwimLane = ({swimLaneData, swimLaneTitle, isLoading}) => {

  return (
      <Box sx={{
        display: 'flex',
        alignItems: 'flex-start',
        flexDirection: 'column',
        bgcolor: '#9bbedb',
        p: 1,
        m: 2,
        borderRadius: 1
      }}
      >
        <Typography>{swimLaneTitle}</Typography>
        <Box sx={{
          display: 'flex',
          flexDirection: 'column',
          p: 1,
          m: 1,
          borderRadius: 3
        }}
      >
        {swimLaneData?.map(card =>
          <KBCard title={card.taskTitle} desc={card.taskDescription} key={card._id}/>
          )
        }
      </Box>
    </Box>

  )
}
export default SwimLane;