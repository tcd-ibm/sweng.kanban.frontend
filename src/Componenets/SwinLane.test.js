import React from 'react';
import renderer from 'react-test-renderer';
import SwimLane from './SwimLane';

it('renders correctly when there are no items', () => {
  const tree = renderer.create(<SwimLane />).toJSON();
  expect(tree).toMatchSnapshot();
});