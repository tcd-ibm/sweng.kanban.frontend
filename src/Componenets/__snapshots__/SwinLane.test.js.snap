// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<div
  className="MuiBox-root MuiBox-root-1"
>
  <p
    className="MuiTypography-root MuiTypography-body1"
  />
  <div
    className="MuiBox-root MuiBox-root-2"
  />
</div>
`;
